flink-clients/src/main/java/org/apache/flink/client/cli/DefaultCLI.java

```diff
@@ -37,16 +38,24 @@
  */
 public class DefaultCLI extends AbstractCustomCommandLine {
 
+	public static final String ID = "default";
+
 	private static final Option addressOption = new Option("m", "jobmanager", true,
 		"Address of the JobManager to which to connect. " +
 			"Use this flag to connect to a different JobManager than the one specified in the configuration. " +
 			"Attention: This option is respected only if the high-availability configuration is NONE.");
 
-	public static final String ID = "default";
-
-	public DefaultCLI(Configuration configuration) {
-		super(configuration);
-	}
+	/**
+	 * Dynamic properties allow the user to specify additional configuration values with -D, such as
+	 * <tt> -Dfs.overwrite-files=true  -Dtaskmanager.memory.network.min=536346624</tt>.
+	 */
```

Comment:
If I am not mistaken, now all `CustomCommandLines` have dynamic options. So why not moving the option and the encode/decode logic to the base class or a utility class (for the encode/decode)? 
Actually, I believe that this will bring the `Generic` and the `Default CLI` codebases even closer.

Reply:
I'll move it to a Util. Unfortunately, the `GenericCLI` and `DefaultCLI` don't have a common base class. And I also don't want to touch the dynamic properties code of the `FlinkYarnSessionCli` which is yet another story... 😅